### PR TITLE
Issue/120/events endpoint app

### LINKED ISSUE #120: Add an event history endpoint to say when something has been published/unpublished.
- [x] Create a script that can diff two elasticsearch indices and generate a list of changes. Your script should:
        - Use command line arguments to provide `source` and `destination` indices
        - Use the collection_id field to find which records have been added and which have been removed.
        - Store the title field so it is easier for a user to understand what has changed
        - Generate a list of dictionary "Events" i.e. ```{ 'collection_id': '', 'collection_title': '', 'action': 'added/removed', 'datetime': 
        '<current_datetime in ISO format'```
- [x] Create a new APP for the events endpoint as part of the archive-opensearch repo
- [x] Create an `Event` model in your new app to hold this information
- [x] Create a POST endpoint using the [django rest framework](https://www.django-rest-framework.org/) to post the information from the script in the first step
- [x] Create a GET endpoint using the django-rest-framework to display the events and add a time filter to allow the user to specify the time interval for getting events. e.g. `GET` /events?start_date=2021-07-25
- [x] Add collection_id filter to GET endpoint to allow the user to `GET` events for a specific collection. i.e. collection added..., collection removed... 
- [x] Integrate POST endpoint into script to complete the pipeline. diff-index -> generate events -> POST to API -> retrieve recent events
- [x] Add authentication to the POST endpoint to protect against random people posting things
- [x] Add additional event check to see if DRS IDs have changed and create an "updated" event if they have

### PR DESCRIPTION
**Pull request by Mahir-Sparkess**

### CODE REVIEW

File: events/views.py [hunk 1]
```diff
@@ -7,39 +8,49 @@
...
-                   APIView,):
+class EventViewSet(
+    GenericViewSet,
+    generics.ListAPIView,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    generics.GenericAPIView,
+    APIView,
+):
+    """A class based view for Events"""
```
Comment (rsmith013) [inline events/views.py]: Add more detailed docstring in here. This is what is displayed against the browseable API.

This docstring needs to explain to a new user what this endpoint is, what it is for and how they can use it. 
Include information about the query parameters and give examples of how to use them.


e.g.
http://api.catalogue.ceda.ac.uk/api/v2/observations/
https://github.com/cedadev/ceda_moles_django/blob/e59d8eb79ff07d993648c3e2b633465d91ccd1e4/cedamoles_api/v2/views/observation.py#L135

**Review by rsmith013 — CHANGES_REQUESTED** [review]

File: events/events_settings.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+REST_FRAMEWORK = {
```
Comment (rsmith013) [inline events/events_settings.py]: I would just put this in the project root settings.py. No need to have an extra import. When the application is deployed in production, the settings file is completely overridden.

**Review by rsmith013 — CHANGES_REQUESTED** [review]

File: events/events_settings.py (hunk 2 above)
Comment (rsmith013) [reply events/events_settings.py]: If you wanted to make the application "re-useable" add a README inside the events app and write down the required SETTINGS

File: events/views.py (hunk 1 above)
Comment (Mahir-Sparkess) [reply events/views.py]: I am unable to see the that repo, I would like to know how to add headings/convention to writing the class docstring, right now following pydoc's convention.

Comment (rsmith013) [reply events/views.py]: image

Comment (rsmith013) [reply events/views.py]: it's just markdown

**Pull request merged**